### DISCUSSION TITLE
UWP and non-Mvvm project?

### DISCUSSION THREAD

**Discussion by NomadicDeveloper22** [Live-Charts/LiveCharts2#485] [Q&A] [answered]
I coming from being an almost Java only developer, and now using uwp to create a company project. We don't really like Mvvm and the project uses "code-behind" I believe its called (or maybe plain old mvc)

I'm still getting used to using libraries with nuget, but would LiveCharts work in our case? There really doesn't seem to be any real alternative to displaying charts

I tried to follow the Uno documentation (as instructed by the uwp docs), and placing a `<lvc:CartesianChart>` in a stackpane and got the following error:
`A value of type 'CartesianChart' cannot be added to a collection or dictionary of type 'UIElementCollection'`

And from what I was able to find through the github issues, is that I need to be using mvvm. Hopefully that's not the case

**Comment by beto-rodriguez** [comment, marked as answer]
> but would LiveCharts work in our case?

I hope it does, that is the idea of building this library.

> A value of type 'CartesianChart' cannot be added to a collection or dictionary of type 'UIElementCollection'

Strange, does your code compile? or is it only an error that VS is showing (it happens sometimes in XAML).

>  is that I need to be using mvvm. Hopefully that's not the case

No mvvm is not necessary, it just helps me to share the same view model for all the platforms, it saves me a lot of time! take a look at the WinForms or EtoForms sample, those platforms do not support mvvm,

**Reply by beto-rodriguez** [reply]
strange, need to check this it is also maybe related to [this](https://github.com/beto-rodriguez/LiveCharts2/discussions/266#discussioncomment-2845064).

**Reply by beto-rodriguez** [reply]
@TheRedSpy15 Glad you made it work! we need to specify that in the docs, it is not anywhere!

**Comment by Strypper** [comment]
Hi what's the solution for this?
